### PR TITLE
feat(build): remove base-requirements.txt

### PR DESCRIPTION
**Pull request by hsheth2**
We want the ingestion-base image to be as slim as possible - that means not installing packages that we aren't sure will be used by images built on top of this one. We had previously needed this to improve build times since `pip` was slow, but our migration to `uv` obviates the need. This should also help slim down the datahub-actions container.

This PR also sets up depot.dev. This helps us avoid using qemu for arm emulation, which slows down our builds significantly.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by david-leifker — CHANGES_REQUESTED** [review]
Unfortunately, this adds significant build time whenever `metadata-ingestion` is changed. Currently this time cost is only paid when the base requirements.txt is updated.

For the slim ingestion image the build runs for [51 minutes](https://github.com/datahub-project/datahub/actions/runs/10564733668/job/29268222840?pr=11238) and for full it runs [1 hour and 8 minutes](https://github.com/datahub-project/datahub/actions/runs/10564733668/job/29268297099?pr=11238).

File: docker/datahub-ingestion/Dockerfile [hunk 1]
```diff
@@ -21,19 +21,20 @@ ARG PIP_MIRROR_URL
...
+WORKDIR /metadata-ingestion
 RUN sed -i.bak "s/__version__ = \"1\!0.0.0.dev0\"/__version__ = \"$(echo $RELEASE_VERSION|sed s/-/+/)\"/" src/datahub/__init__.py && \
     sed -i.bak "s/__version__ = \"1\!0.0.0.dev0\"/__version__ = \"$(echo $RELEASE_VERSION|sed s/-/+/)\"/" airflow-plugin/src/datahub_airflow_plugin/__init__.py && \
     cat src/datahub/__init__.py | grep __version__ && \
     cat airflow-plugin/src/datahub_airflow_plugin/__init__.py | grep __version__
 
 FROM base AS slim-install
 
-RUN uv pip install --no-cache -e ".[base,datahub-rest,datahub-kafka,snowflake,bigquery,redshift,mysql,postgres,hive,clickhouse,glue,dbt,looker,lookml,tableau,powerbi,superset,datahub-business-glossary]"
+RUN --mount=type=cache,target=/datahub-ingestion/.cache/uv,uid=1000,gid=1000 \
```
Comment (david-leifker) [inline docker/datahub-ingestion/Dockerfile:36]: Curious as to why this one is still `/datahub-ingestion` while it seems to be changed everywhere else?

Comment (david-leifker) [reply docker/datahub-ingestion/Dockerfile:36]: Maybe this is separating the `metadata-ingestion` source from the user's home directory

Comment (hsheth2) [reply docker/datahub-ingestion/Dockerfile:36]: Yup that's it
- `/datahub-ingestion` is the user's home directory - there's nothing in here other than dotfiles and the uv cache
- `/metadata-ingestion` has the code from the metadata-ingestion directory

I can make it restore the workdir

**Review by david-leifker — APPROVED** [review]

**Pull request merged**